### PR TITLE
NIM-7 Update JS to pass JSON for /new

### PR DESCRIPTION
**Pull request by dodecaphonix**
Update JavaScript to pass JSON for requests to /new
Add POST method to /new
Add UI elements to get values of

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -16,12 +16,6 @@
 
 @app.route('/')
 def index():
-    json = loads('{"min": 3, "max": 12, "piles": 3}')
```
Comment (dodecaphonix) [inline app.py]: This was dead code, so I deleted it.

File: app.py [hunk 2]
```diff
@@ -38,8 +32,8 @@ def update_game_state():
     return jsonify({'state': state})
 
 
-@app.route('/new')
-def newgame():
+@app.route('/new', methods=['GET', 'POST'])
```
Comment (dodecaphonix) [inline app.py]: Added POST so that my changes wouldn't break the app

File: templates/index.html [hunk 3]
```diff
@@ -3,34 +3,64 @@
   <head>
     <title>Play Nim</title>
     <script src="//ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
+    <style>
```
Comment (daveLambe) [inline templates/index.html]: As @romainrichard suggested, please include screenshot when doing frontend stuff 😜

File: static/game_operations.js [hunk 4]
```diff
@@ -3,7 +3,18 @@ var maxPiles = 5;
...
 
 function newGame() {
-  $.getJSON( "/new", updatePage);
+  $.ajax({
+    type:"POST",
+    url: "/new",
+    data: JSON.stringify({
+      "min": $('#min').val(),
+      "max": $('#max').val(),
+      "piles": $('#piles').val()
```
[the remark this answers is not in the document]
Comment (daveLambe) [reply static/game_operations.js]: +1

File: static/game_operations.js [hunk 5]
```diff
@@ -3,7 +3,18 @@ var maxPiles = 5;
...
+    }),
+    contentType: "application/json; charset=utf-8",
+    dataType : 'json',
+    success: updatePage
```
Comment (daveLambe) [inline static/game_operations.js]: should we also have an `error: ` here?

File: templates/index.html [hunk 6]
```diff
@@ -3,34 +3,64 @@
   <head>
     <title>Play Nim</title>
     <script src="//ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
```
Comment (daveLambe) [inline templates/index.html]: An example of using AJAX with JQuery for Flask includes a line to also put the Google Library in the project static folder for 'fallback'. 
Do you think that might be useful here? 

The example is: 
```
<script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script>window.jQuery || document.write('<script src="{{
  url_for('static', filename='jquery.js') }}">\x3C/script>')</script>
```

Comment (dodecaphonix) [reply templates/index.html]: Yes, that's absolutely a good idea. We don't want our site to go down just because `ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js` did, but we still want to steal their bandwidth when we can.

File: static/game_operations.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (dodecaphonix) [reply static/game_operations.js]: To the HTML IDs or the JSON?

File: templates/index.html [hunk 7]
```diff
@@ -3,34 +3,64 @@
...
+    <style>
+      body {
+        background-color: #EEEEEE;
+      }
+      div {
+        color: #555555;
+        padding: 10px 10px 10px 10px;
+      }
+    </style>
```
[the remark this answers is not in the document]
Comment (dodecaphonix) [reply templates/index.html]: Merci beaucoup! :3

File: templates/index.html [hunk 8]
```diff
@@ -3,34 +3,64 @@
...
+      <button onclick="newGame()">New Game</button>
+    </div>
 
-    <br><br>
+    <div>
+      <select id="piles">
+        <option value=3>3</option>
+        <option value=4>4</option>
+        <option value=5>5</option>
+      </select>
```
[the remark this answers is not in the document]
Comment (dodecaphonix) [reply templates/index.html]: Doing what? The front end has a finite number of elements that we can populate with piles. I can make any number of them. I just chose 3 to 5 because that's what I generally end up playing in real life situations. Were you asking for a lower min, higher max, or to let them choose any number?

File: static/game_operations.js (hunk 5 above)
Comment (dodecaphonix) [reply static/game_operations.js]: We could. I was holding off until we know what we'd want to do in the case of as error.

File: static/game_operations.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (romainrichard) [reply static/game_operations.js]: Nevermind, we'll address this issue in https://hacksaurz.atlassian.net/browse/NIM-23

File: templates/index.html (hunk 8 above)
[the remark this answers is not in the document]
Comment (romainrichard) [reply templates/index.html]: Let us choose any number. If later on we realize that it doesn't make sense and we need to limit it, then sure. But for now we don't need to limit ourselves.

[the remark this answers is not in the document]
Comment (dodecaphonix) [reply templates/index.html]: Hmmm... The jQuery is updating a finite number of HTML elements. It locates them by ID. Should I add an infinite number of HTML tags? See the `id="pile0"` in `<div style="float: left;"><h1 id="pile0" ></h1></div>`.

[the remark this answers is not in the document]
Comment (romainrichard) [reply templates/index.html]: I have no idea how to do it, but there has to be a way to make it work with regular text boxes.

[the remark this answers is not in the document]
Comment (dodecaphonix) [reply templates/index.html]: This is a bigger ask than you realize. We can always `curl` if we want to test extreme cases. Let's file a ticket for this work and settle on a min and a max for now.

[the remark this answers is not in the document]
Comment (dodecaphonix) [reply templates/index.html]: Disregard previous comment. I've implemented this.

[the remark this answers is not in the document]
Comment (romainrichard) [reply templates/index.html]: As I just mentioned on Slack, I was only asking you to revert the min/max/piles thing so they were text boxes. Changing `updatePage` wasn't necessary but I'm glad it's done, now I believe we have a fully functional UI thanks to you :tada:

File: static/jquery.js [hunk 9]
```diff
@@ -0,0 +1,5 @@
+/*! jQuery v2.1.4 | (c) 2005, 2015 jQuery Foundation, Inc. | jquery.org/license */
```
Comment (romainrichard) [inline static/jquery.js]: Why is this file needed?

Comment (dodecaphonix) [reply static/jquery.js]: jQuery is a third party JavaScrip library. This is just a backup of the jQuery library. The client needs to get this library in order to work. Now, if the client can't get it for some reason, we can give it to them ourselves. You can spin up the server, turn off wifi, and comment out the following code to see it break. Then, add the following code back in to see it work.
```
    <script>
      window.jQuery || document.write(
        '<script src="{{ url_for('static', filename='jquery.js') }}">\x3C/script>'
      )
    </script>
```
All this code is is an `or` statement. "Does the window already have `window.jQuery`? If not, get it from `static/jquery.js`."

Comment (romainrichard) [reply static/jquery.js]: I see. Is it common to do this?

Comment (dodecaphonix) [reply static/jquery.js]: Very common. It's my bad that I didn't do it in the first place. Imagine if clients had to pull all of the dependancies from pipy every time. It would be pretty bad if pipy went down.

File: templates/index.html [hunk 10]
```diff
@@ -2,7 +2,15 @@
...
   <head>
     <title>Play Nim</title>
+
     <script src="//ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
+
+    <script>
+      window.jQuery || document.write(
+        '<script src="{{ url_for('static', filename='jquery.js') }}">\x3C/script>'
+      )
+    </script>
```
Comment (daveLambe) [inline templates/index.html]: I see this now. Completely my bad, I thought it wasn't added
